web-server/v0.4/src/components/TableTree/index.js

```diff
@@ -0,0 +1,111 @@
+import React, { PureComponent } from 'react';
+import PropTypes from 'prop-types';
+import { Table } from 'antd';
+
+export default class TableTree extends PureComponent {
+  static propTypes = {
+    id: PropTypes.string,
+    dataSource: PropTypes.array,
+    extension: PropTypes.array,
+    onload: PropTypes.func,
+    config: PropTypes.object,
+  };
+
+  static defaultProps = {
+    id: '',
+    dataSource: [],
+    extension: [],
+    onload: () => {},
+    config: {},
+  };
+
+  constructor(props) {
+    super(props);
+    this.fileData = [];
+    this.state = {
+      expandedKeys: [],
+    };
+  }
+
+  componentDidMount() {
+    const { dataSource, onload } = this.props;
+    onload(dataSource);
+  }
+
+  onExpand = (expanded, record) => {
+    let { expandedKeys } = this.state;
+    const keys = expandedKeys;
+    expandedKeys = expanded ? keys.concat(record.key) : keys.slice(0, keys.indexOf(record.key));
+    this.setState({ expandedKeys });
+  };
+
+  filterNames = (record, value) =>
+    record.some(childRecord => {
+      if (childRecord.name.toLowerCase().includes(value)) {
+        return true;
+      }
+      if (childRecord.children) {
+        return this.filterNames(childRecord.children, value);
+      }
+      return false;
+    });
+
+  render() {
```

Comment:
`TableTree` only visualizes 10 'hits' from Elasticsearch as the query size defaults to 10. 

@portante We're seeing up to 88k hits from one result parent and querying them all would be bad for performance. Is it possible to query directory information progressively as users click through different levels of the file tree?

Reply:
@Anisha1234 Can you see if the scan & scroll API offered by elasticsearch solves this issue for query size? https://www.elastic.co/guide/en/elasticsearch/guide/1.x/scan-scroll.html